.github/workflows/python_wheels.yml

```diff
@@ -216,6 +216,24 @@ jobs:
               fi
             done
 
+      - name: Run Python MPI tests
+        if: matrix.os_image == 'ubuntu:22.04'
```

Comment:
I've modified this step:

-  `redhat/ubi9:9.2` is the default base image 

- Install openmpi and mpi4py via conda. The install script is taken from the PyPI's README.md.

This is the revised CI script: https://github.com/1tnguyen/cuda-quantum/blob/74943d3754f323b2ec70e0760853fa5658adcfe7/.github/workflows/python_wheels.yml#L219-L261 

I skipped the CUDA install step since it is not required for this test.

It ran successfully for both [amd64](https://github.com/NVIDIA/cuda-quantum/actions/runs/7054568601/job/19204129505?pr=966) and [arm64](https://github.com/NVIDIA/cuda-quantum/actions/runs/7054568601/job/19204090802?pr=966).